src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,45 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INITIAL_LENGTH = 10;
+    private V[] values;
+    private K[] keys;
+    private int size;
+
+    public StorageImpl() {
+        values = (V[]) new Object[INITIAL_LENGTH];
+        keys = (K[])new Object[INITIAL_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (compareKeys(this.keys[i], key)) {
+                this.values[i] = value;
+                return;
+            }
+        }
+        this.keys[size] = key;
+        this.values[size] = value;
```

Comment:
```suggestion
            if (compareKeys(keys[i], key)) {
                values[i] = value;
                return;
            }
        }
        keys[size] = key;
        values[size] = value;
```
avoid `this` if there is no necessity in it
code is easier to read